graspy/embed/oosase.py

```diff
@@ -0,0 +1,416 @@
+# oosase.py
+# Created by Hayden Helm on 2019-04-22.
+# Email: hhelm2@jhu.edu
+import warnings
+
+from .base import BaseEmbed
+from .svd import selectSVD
+from ..utils import import_graph, get_lcc, is_fully_connected, is_symmetric
+
+import numpy as np
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils import check_array
+import networkx as nx
+
+
+class OutOfSampleAdjacencySpectralEmbed(BaseEmbed):
+    r"""
+    Class for computing the out of sample adjacency spectral embedding of a graph.
+    
+    The adjacency spectral embedding (ASE) is a k-dimensional Euclidean representation of 
+    the graph based on its adjacency matrix [1]_. It relies on an SVD to reduce the dimensionality
+    to the specified k, or if k is unspecified, can find a number of dimensions automatically
+    (see graspy.embed.selectSVD). The out of sample adjacency spectral embedding (OOSASE) considers
+    the ASE of an induced subgraph of the original graph. To embed "out of sample" vertices, a projection
+    matrix learned from the in sample embedding is used [2]_.
+
+    Parameters
+    ----------
+    n_components : int or None, default = None
+        Desired dimensionality of output data. If "full", 
+        n_components must be <= min(X.shape). Otherwise, n_components must be
+        < min(X.shape). If None, then optimal dimensions will be chosen by
+        ``select_dimension`` using ``n_elbows`` argument.
+    n_elbows : int, optional, default: 2
+        If `n_components=None`, then compute the optimal embedding dimension using
+        `select_dimension`. Otherwise, ignored.
+    algorithm : {'randomized' (default), 'full', 'truncated'}, optional
+        SVD solver to use:
+
+        - 'randomized'
+            Computes randomized svd using 
+            ``sklearn.utils.extmath.randomized_svd``
+        - 'full'
+            Computes full svd using ``scipy.linalg.svd``
+        - 'truncated'
+            Computes truncated svd using ``scipy.sparse.linalg.svd``
+    n_iter : int, optional (default = 5)
+        Number of iterations for randomized SVD solver. Not used by 'full' or 
+        'truncated'. The default is larger than the default in randomized_svd 
+        to handle sparse matrices that may have large slowly decaying spectrum.
+    check_lcc : bool , optional (default = True)
+        Whether to check if input graph is connected. May result in non-optimal 
+        results if the graph is unconnected. If True and input is unconnected,
+        a UserWarning is thrown. Not checking for connectedness may result in 
+        faster computation.
+    in_sample_proportion : float , optional (default = 1)
+        If in_sample_idx is None, the proportion of in sample vertices to use for
+        initial embedding.
+    in_sample_idx : array-like , optional (default = None)
+    connected_attempts : integer , optional (default = 1000)
```

Comment:
incorrect default

Reply:
que ?

Reply:
the default in the init is 100, here you say it is 1000